src/stack.rs

```diff
@@ -53,6 +53,43 @@ impl Stack {
     pub(crate) fn clear_pointer_flag(&mut self, slot: u16) {
         self.pointer_flags.clear(slot);
     }
+
+    pub(crate) fn snapshot(&self) -> StackSnapshot {
+        let mut dirty_prefix_end = 0;
+        for i in 0..NUMBER_OF_DIRTY_AREAS {
+            if self.dirty_areas & (1 << i) != 0 {
+                dirty_prefix_end = i + 1;
+            }
+        }
```

Comment:
IIUC, this can be optimized using [`leading_zeros`](https://doc.rust-lang.org/std/primitive.u64.html#method.leading_zeros) which translates to a single assembly instruction on x86.

Reply:
I believe LLVM will remove the loop anyway but this reminds me that I was thinking of maybe saving the dirty parts instead of the dirty prefix. WDYT about that?

Reply:
Saving dirty parts looks more logical given that we track them, but I'm not sure about performance impact of this decision. Regarding LLVM, it will probably unroll the loop, but I'm not sure it's smart enough to detect simplification to a  "count leading zeros" instruction.

Reply:
I verified with godbolt.org. This is indeed a case where LLVM doesn't realize it can emit `bsr`.